splash/qtrender_lua.py

```diff
@@ -452,7 +452,21 @@ def set_custom_headers(self, headers):
     def set_viewport(self, size):
         if size is None:
             return
-        return self.tab.set_viewport(size)
+        return self.tab.set_viewport(size, raise_if_empty=True)
+
+    @command()
+    def get_viewport_size(self):
+        sz = self.tab.web_page.viewportSize()
+        return sz.width(), sz.height()
+
+    @command()
+    def get_window_size(self):
+        sz = self.tab.web_view.size()
+        return sz.width(), sz.height()
+
+    @command()
+    def set_window_size(self, sz):
+        self.tab.set_window_size(sz)
```

Comment:
I'm not a fan of forwarding QSize values into Lua runtime (C++ value wrapped as Python value wrapped as Lua value, eh?). I guess it should return `WxH` strings then to roundtrip successfully, it feels odd but could work.